### PR TITLE
Add test leaf template with serverAuth EKU

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
This change introduces `leaf00.cert` and `leaf00.chain`, a template cert/chain with the `serverAuth` EKU.

It was generated using `$ make leaf00.cert leaf00.chain` command.

```
$ openssl x509 -in leaf00.cert -text
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number: 3735928559 (0xdeadbeef)
        Signature Algorithm: ecdsa-with-SHA256
        Issuer: C = GB, ST = London, L = London, O = Google, OU = Eng, CN = FakeIntermediateAuthority
        Validity
            Not Before: Mar 18 15:39:22 2022 GMT
            Not After : Apr 30 15:39:22 2029 GMT
        Subject: C = GB, ST = London, O = Google, OU = Eng, CN = leaf00.csr.pem
        Subject Public Key Info:
            Public Key Algorithm: id-ecPublicKey
                Public-Key: (256 bit)
                pub:
                    04:eb:37:4e:52:45:9c:46:d5:a8:b8:c5:ed:58:b9:
                    30:29:a6:70:8a:69:a0:26:5c:9e:2f:6e:b8:6b:23:
                    6c:84:e1:46:3a:98:36:82:44:a5:8a:17:8b:41:82:
                    32:f4:2d:e0:08:5b:7e:07:38:52:fc:47:56:28:27:
                    9b:ed:60:8b:ac
                ASN1 OID: prime256v1
                NIST CURVE: P-256
        X509v3 extensions:
            X509v3 Subject Key Identifier: 
                3F:B2:2F:41:FC:11:9A:D3:8D:A6:85:80:84:86:AE:7E:73:2E:69:5D
            X509v3 Authority Key Identifier: 
                keyid:05:06:07:08

            X509v3 Key Usage: critical
                Digital Signature, Non Repudiation, Key Encipherment, Data Encipherment, Key Agreement, Encipher Only, Decipher Only
            X509v3 Extended Key Usage: 
                TLS Web Server Authentication
    Signature Algorithm: ecdsa-with-SHA256
         30:44:02:20:33:c6:78:70:56:30:0b:b8:34:72:19:ed:ad:f3:
         c9:b7:1f:02:ff:d9:25:f0:21:63:db:fe:e2:24:3b:be:8e:95:
         02:20:06:e6:7d:87:13:e6:2e:9d:20:3f:82:b1:02:c8:e4:fa:
         6f:3d:fe:b2:a0:e8:2f:5f:74:aa:76:0b:23:5e:1c:e5
```

### Checklist

- [ ] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [ ] I have updated [documentation](docs/) accordingly.

### CODE REVIEW

File: trillian/testdata/int-ca.cfg [hunk 1]
```diff
@@ -23,6 +23,12 @@ subjectKeyIdentifier = hash
 authorityKeyIdentifier = keyid:always,issuer
 keyUsage = critical, digitalSignature, nonRepudiation, keyEncipherment, dataEncipherment, keyAgreement, encipherOnly, decipherOnly
 
+[ v3_user_serverAuth ]
+subjectKeyIdentifier = hash
+authorityKeyIdentifier = keyid:always,issuer
+keyUsage = critical, digitalSignature, nonRepudiation, keyEncipherment, dataEncipherment, keyAgreement, encipherOnly, decipherOnly
```
Comment (AlCutter) [inline trillian/testdata/int-ca.cfg:29]: Nit/curiosity: was it intentional to have all these extra KU on here, or would `digitalSignature` be enough?

**Review by AlCutter — APPROVED** [review]

File: trillian/testdata/int-ca.cfg (hunk 1 above)
Comment (pavelkalinnikov) [reply trillian/testdata/int-ca.cfg:29]: Not intentional, I just copy-pasted the `v3_user` option set from above, and added the EKU.

**Pull request merged**